includes/class-wc-calypso-bridge-free-trial.php

```diff
@@ -0,0 +1,257 @@
+
+			return $value;
+		}, PHP_INT_MAX );
+
+		/****** PAYPAL Express Checkout ******/
+		add_filter( 'pre_option_woocommerce-ppcp-settings', '__return_false', PHP_INT_MAX );
+		// There is no need for a `default_option` filter, as the default value is an empty array.
+		add_filter( 'option_woocommerce-ppcp-settings', function ( $value ) {
+
+			// Bail out early if the current user is allowed to create orders on free trial.
+			if ( current_user_can( 'manage_woocommerce' ) ) {
+				return $value;
+			}
+
+			$value['button_product_enabled']      = false;
+			$value['button_cart_enabled']         = false;
+			$value['button_mini-cart_enabled']    = false;
+			$value['pay_later_button_enabled']    = false;
+			$value['pay_later_messaging_enabled'] = false;
+
+			return $value;
+		}, PHP_INT_MAX );
+
+		// Disable PayPal's OXXO gateway for non-admin users.
+		add_filter( 'pre_option_woocommerce_ppcp-oxxo-gateway_settings', '__return_false', PHP_INT_MAX );
+		add_filter( 'option_woocommerce_ppcp-oxxo-gateway_settings', function ( $value ) {
+
+			// Bail out early if the current user is allowed to create orders on free trial.
+			if ( current_user_can( 'manage_woocommerce' ) ) {
+				return $value;
+			}
+
+			$value['enabled'] = 'no';
+
+			return $value;
+		}, PHP_INT_MAX );
+
+		// Disable PayPal's Pay Upon Invoice gateway for non-admin users.
+		add_filter( 'pre_option_woocommerce_ppcp-pay-upon-invoice-gateway_settings', '__return_false', PHP_INT_MAX );
+		add_filter( 'option_woocommerce_ppcp-pay-upon-invoice-gateway_settings', function ( $value ) {
+
+			// Bail out early if the current user is allowed to create orders on free trial.
+			if ( current_user_can( 'manage_woocommerce' ) ) {
+				return $value;
+			}
+
+			$value['enabled'] = 'no';
+
+			return $value;
+		}, PHP_INT_MAX );
+
+	}
+
+	/**
+	 * Frontend free trial hooks.
+	 */
+	public function frontend() {
+
+		// Bail out early if the current user is allowed to create orders on free trial.
+		if ( current_user_can( 'manage_woocommerce' ) ) {
+			return;
+		}
+
+		// Prevent orders on shortcode checkout.
+		add_action( 'woocommerce_before_checkout_process', function () {
+			throw new Exception( __( 'Your order could not be placed. Checkout functionality is currently enabled for preview purposes only.', 'wc-calypso-bridge' ) );
+		}, PHP_INT_MAX );
+
+		// Prevent orders on shortcode checkout - PayPal removes the checkout button and replaces it.
+		add_action( 'woocommerce_after_checkout_validation', function ( $data, $errors ) {
+			$errors->add(
+				405,
+				__( 'Your order could not be placed. Checkout functionality is currently enabled for preview purposes only.', 'wc-calypso-bridge' )
+			);
+		}, PHP_INT_MAX, 2 );
+
+		// Prevent orders on block checkout.
+		add_action( 'woocommerce_store_api_checkout_order_processed', function () {
+			throw new Automattic\WooCommerce\StoreApi\Exceptions\RouteException(
+				405,
```

Comment:
Nit: I suggest using something more generic here as an error code. For example, `400 Bad Request` or a `409 Conflict`. 

Store API, by default, uses the `409` when there is an error in the checkout.